Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,841 @@
+        for (MarkerInfo info : markersInfos) {
+            updateMarker(info, isDraggable);
+        }
+    }
+
+    @Override
+    public Set<MarkerInfo> getMarkerInfoList() {
+        return new HashSet<MarkerInfo>(mBiMarkersMap.keySet());
+    }
+
+    @Override
+    public List<LatLong> projectPathIntoMap(List<LatLong> path) {
+        List<LatLong> coords = new ArrayList<LatLong>();
+        Projection projection = getBaiduMap().getProjection();
+
+        for (LatLong point : path) {
+            LatLng coord = projection.fromScreenLocation(new Point((int) point
+                    .getLatitude(), (int) point.getLongitude()));
+            coords.add(LatLngToCoord(coord));
+        }
+
+        return coords;
+    }
+
+    @Override
+    public void removeMarkers(Collection<MarkerInfo> markerInfoList) {
+        if (markerInfoList == null || markerInfoList.isEmpty()) {
+            return;
+        }
+
+        for (MarkerInfo markerInfo : markerInfoList) {
+            Marker marker = mBiMarkersMap.getValue(markerInfo);
+            if (marker != null) {
+                marker.remove();
+                mBiMarkersMap.removeKey(markerInfo);
+            }
+        }
+    }
+
+    @Override
+    public void setMapPadding(int left, int top, int right, int bottom) {
+        getBaiduMap().setViewPadding(left, top, right, bottom);
+    }
+
+    @Override
+    public void setOnMapClickListener(OnMapClickListener listener) {
+        mMapClickListener = listener;
+    }
+
+    @Override
+    public void setOnMapLongClickListener(OnMapLongClickListener listener) {
+        mMapLongClickListener = listener;
+    }
+
+    @Override
+    public void setOnMarkerDragListener(OnMarkerDragListener listener) {
+        mMarkerDragListener = listener;
+    }
+
+    @Override
+    public void setOnMarkerClickListener(OnMarkerClickListener listener) {
+        mMarkerClickListener = listener;
+    }
+
+    @Override
+    public void setLocationListener(android.location.LocationListener receiver) {
+        mLocationListener = receiver;
+        if (mBDLocListener != null) {
+            BDLocation bdloc = mBDLocClient.getLastKnownLocation();
+            updateBDMapStatus(bdloc);
+            notifyToLocatorActivity(bdloc);
+        }
+    }
+
+    private void updateCamera(final LatLong coord) {
+		final BaiduMap map = getBaiduMap();
+        if ( map != null && coord != null) {
+			final float zoomLevel = map.getMapStatus().zoom;
+			map.animateMapStatus(MapStatusUpdateFactory.newLatLngZoom(CoordToLatLng(coord), zoomLevel));
+		}
+    }
+
+    @Override
+    public void updateCamera(final LatLong coord, final float zoomLevel) {
+		final BaiduMap map = getBaiduMap();
+        if ( map != null && coord != null) {
+            map.animateMapStatus(MapStatusUpdateFactory.newLatLngZoom(CoordToLatLng(coord), zoomLevel));
+        }
+    }
+
+    @Override
+    public void updateCameraBearing(float bearing) {
+       return; // No bearing in BaiduMap yet.
```

Comment:
There are "overlook" and "rotate" fields in class MapStatus in Baidumap, I'm not sure which one is the equivalent of bearing in GoogleMap.